Transfer kpoints to sirius v2

toxa81 opened the pull request.

gcistaro committed:
Error in Cholesky factorization

Commit:
creating new kset and ground state for nscf

toxa81 committed:
fixes

gcistaro committed:
Moving nscf to do_nscf, getting H0 for each q

gcistaro committed:
Moving nscf, calculating H0 for each iq

gcistaro committed:
Fixing error in kcw_run_nscf

gcistaro committed:
enabling number of steps

gcistaro committed:
Removing error files from branch

gcistaro committed:
removing unused variable in kcw_run_nscf

gcistaro committed:
enabling nscf in HP

toxa81 committed:
try 200 steps

gcistaro committed:
temporarely disabling SIRIUS nscf in HP for checking ci-tests

gcistaro committed:
temporarely disabling SIRIUS nscf in HP for checking ci-tests --- correct typo

gcistaro committed:
Fixing error from setup_sirius when nscf is not enabled

toxa81 committed:
try without new handlers

toxa81 commented:
cscs-ci run default

toxa81 committed:
cleanup

toxa81 commented:
@gcistaro Looking at your modifications I realized that the new handlers are not neeed, so I removed them. CI/CD passes. I think this is good to be merged. TODO for later: understand why in CrI3 running nscf with Sirius leads to different results.

gcistaro commented:
About CrI3, can it be related to PAW pseudos? It is the only example running HP with PAW.

gcistaro committed:
Fixing reading of v>1 upf file, when n is not specified in it

gcistaro committed:
remove blank line added by mistake

gcistaro committed:
Merge branch 'fix/read-upf' into Transfer_kpoints_to_SIRIUS_v2

gcistaro committed:
adding header when nscf is done with SIRIUS

gcistaro committed:
Fixing issue with MPI and the et array. Now et is filled with the right eigenvalues.

gcistaro committed:
Fixing eigenvalues storage in qe. et needs all the eigenvalues for rank=0, only local eigenvalues for rank!=0;
Simplifying get_wave_functions_from_sirius: now the array k_index_map is not needed and we added an if to understand if the k point in the loop is local

gcistaro committed:
enabling nscf in HP

gcistaro committed:
restoring previous version of get_wave_functions_from_sirius

gcistaro committed:
restarting CI

gcistaro committed:
Fixing inconsistency in get_wave_functions (it was now defined without arguments

gcistaro committed:
adding parallel testing for KCW

gcistaro committed:
ask 4 nodes for H2O parallel test
